From 67cee916a0c8f2e7d6274bc0352d5b6bac7b923b Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Wed, 27 Dec 2017 18:50:41 +0800
Subject: [PATCH 2133/2241] dts: a8k: mcbin: bind PCIe comphy lanes for mcbin
 setups

To support PCIe serdes initialization in kernel, it is
needed to bind the COMPHY lanes to corresponding PCIe
port in DTS.

Change-Id: I4a931e38a9c16aae3f6367890db774befb269a80
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/48165
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts | 6 ++++++
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts             | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
index 4bea9cc..b92ad00 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
@@ -205,6 +205,12 @@
 			num-lanes = <4>;
 			status = "okay";
 			reset-gpios = <&cp0_gpio1 20 GPIO_ACTIVE_HIGH>;
+			phys = <&cp0_comphy 0 COMPHY_PCIE0
+				&cp0_comphy 1 COMPHY_PCIE0
+				&cp0_comphy 2 COMPHY_PCIE0
+				&cp0_comphy 3 COMPHY_PCIE0
+				>;
+			phy-names = "pcie-phy0", "pcie-phy1", "pcie-phy2", "pcie-phy3";
 			pinctrl-names = "default";
 			pinctrl-0 = <&pcie_reset_pins>;
 		};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index f2e7c9d..fc5b37c 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -170,6 +170,12 @@
 			num-lanes = <4>;
 			status = "okay";
 			reset-gpios = <&cp0_gpio1 20 GPIO_ACTIVE_HIGH>;
+			phys = <&cp0_comphy 0 COMPHY_PCIE0
+				&cp0_comphy 1 COMPHY_PCIE0
+				&cp0_comphy 2 COMPHY_PCIE0
+				&cp0_comphy 3 COMPHY_PCIE0
+				>;
+			phy-names = "pcie-phy0", "pcie-phy1", "pcie-phy2", "pcie-phy3";
 			pinctrl-names = "default";
 			pinctrl-0 = <&pcie_reset_pins>;
 		};
-- 
2.7.4

